AK/StringUtils.cpp

```diff
@@ -59,6 +59,9 @@ bool matches(StringView str, StringView mask, CaseSensitivity case_sensitivity,
         case '?':
             record_span(string_ptr - string_start, 1);
             break;
+        case '\\':
+            ++mask_ptr;
+            [[fallthrough]];
```

Comment:
I think this needs to break rather than fallthrough. If `\` is the last character in the mask, we'll read off the end of the string.